### ISSUE TITLE
support uncompressed RLE segmentation in datasets

### ISSUE THREAD

**Issue by YQYI** [facebookresearch/detectron2#1692]
I only want to use person instance in coco dataset , so i try to grep instance when register coco data ,I try this code, but get error 
```python
Original Traceback (most recent call last):
  File "/root/miniconda3/envs/pointrender/lib/python3.6/site-packages/torch/utils/data/_utils/worker.py", line 178, in _worker_loop
    data = fetcher.fetch(index)
  File "/root/miniconda3/envs/pointrender/lib/python3.6/site-packages/torch/utils/data/_utils/fetch.py", line 44, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/root/miniconda3/envs/pointrender/lib/python3.6/site-packages/torch/utils/data/_utils/fetch.py", line 44, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/root/miniconda3/envs/pointrender/lib/python3.6/site-packages/detectron2/data/common.py", line 41, in __getitem__
    data = self._map_func(self._dataset[cur_idx])
  File "/root/miniconda3/envs/pointrender/lib/python3.6/site-packages/detectron2/utils/serialize.py", line 23, in __call__
    return self._obj(*args, **kwargs)
  File "/root/miniconda3/envs/pointrender/lib/python3.6/site-packages/detectron2/data/dataset_mapper.py", line 130, in __call__
    for obj in dataset_dict.pop("annotations")
  File "/root/miniconda3/envs/pointrender/lib/python3.6/site-packages/detectron2/data/dataset_mapper.py", line 131, in <listcomp>
    if obj.get("iscrowd", 0) == 0
  File "/root/miniconda3/envs/pointrender/lib/python3.6/site-packages/detectron2/data/detection_utils.py", line 230, in transform_instance_annotations
    mask = mask_util.decode(segm)
  File "/root/miniconda3/envs/pointrender/lib/python3.6/site-packages/pycocotools/mask.py", line 91, in decode
    return _mask.decode([rleObjs])[:,:,0]
  File "pycocotools/_mask.pyx", line 146, in pycocotools._mask.decode
  File "pycocotools/_mask.pyx", line 128, in pycocotools._mask._frString
TypeError: Expected bytes, got list


Process finished with exit code 1
```

my code:
```python
def get_coco_dicts(img_dir, json_path):
    coco_content = grep_coco_dicts(json_path)
    imgs_anns = coco_content['annotations']
    imgs_info = coco_content['images']
    dataset_dicts = {}
    for i in imgs_info:
        record = {}
        record["file_name"] = os.path.join(img_dir, i['file_name'])
        record["image_id"] = i['id']
        record["height"] = i['height']
        record["width"] = i['width']
        record["annotations"] = []
        dataset_dicts[record['image_id']] = record
    for idx, v in enumerate(imgs_anns):
        if v['image_id'] not in dataset_dicts:
            raise Exception(f'标注的image_id不在图片的id集合内{imgs_anns["image_id"]}')
        else:
            obj = {
                "bbox": v['bbox'],
                "bbox_mode": BoxMode.XYXY_ABS,
                "segmentation": v['segmentation'],
                "category_id": 1,
            }
            dataset_dicts[v['image_id']]['annotations'].append(obj)
    result = dataset_dicts
    greped_result = []
    for key, v in result.items():
        if len(v['annotations'])==0:
            pass
        else:
            greped_result.append(v)
    return greped_result



def grep_coco_dicts(json_path, id_needed = 1):
    json_file = json_path
    with open(json_file) as f:
        imgs_anns = json.load(f)
    annotations = imgs_anns['annotations']
    annotations_result = []
    for index, value in enumerate(annotations):
        if value['category_id']!=id_needed:
            pass
        else:
            annotations_result.append(value)
    imgs_anns['annotations'] = annotations_result
    return imgs_anns
```

**Comment by ppwwyyxx** [comment]
Adding `"iscrowd": v['iscrowd'],` should fix it. 
We'll add code to support the uncompressed RLE segmentation in COCO json file as well.

**Comment by YQYI** [comment]
thanks so so so so so so much!!! I just follow this https://detectron2.readthedocs.io/tutorials/datasets.html#register-a-coco-format-dataset, but can not find the error reason.

**Comment by YQYI** [comment]
@ppwwyyxx thanks so so so so so so much!!! I just follow this https://detectron2.readthedocs.io/tutorials/datasets.html#register-a-coco-format-dataset, but can not find the error reason.

**Comment by ppwwyyxx** [comment]
The reason is because COCO's json format does not follow https://detectron2.readthedocs.io/tutorials/datasets.html#standard-dataset-dicts

**Comment by ppwwyyxx** [comment]
Closed by 63f11718c68f1ae951caee157b4e10fae4d7e4be.
`load_coco_json` should be used instead of `json.load` to obtain the correct format that follows https://detectron2.readthedocs.io/tutorials/datasets.html#standard-dataset-dicts.
